File: src/contests/cg/puzzle/go/google-interview---the-two-egg-problem.go
// 🎮 CodinGame Puzzle - google-interview---the-two-egg-problem
// https://www.codingame.com/training/hard/google-interview---the-two-egg-problem

package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	var n int64
	fmt.Fscan(reader, &n)
	var k int64 = 0
	for k*(k+1)/2 < n {
		k++
	}
	fmt.Println(k)
}

File: src/contests/cg/puzzle/perl/google-interview---the-two-egg-problem.pl
# 🎮 CodinGame Puzzle - google-interview---the-two-egg-problem
# https://www.codingame.com/training/hard/google-interview---the-two-egg-problem

my $n = <STDIN>;
chomp $n;
my $k = 0;
$k++ while ($k * ($k + 1) / 2 < $n);
print "$k\n";
